This fixes a number of build issues in r1837:

- LibDrawPin now needs the parent to be passed
- type inconsistency for g_PlotScaleX and g_PlotScaleY between declaration and
  definition
- g_DrawDefaultLineThickness vanished

---

Index: kicad/eeschema/pinedit.cpp
===================================================================
--- kicad.orig/eeschema/pinedit.cpp	2009-06-25 05:36:52.000000000 -0300
+++ kicad/eeschema/pinedit.cpp	2009-06-25 05:36:52.000000000 -0300
@@ -557,7 +557,7 @@
     for( ; DrawItem != NULL; DrawItem = DrawItem->Next() )
         DrawItem->m_Flags = 0;
 
-    CurrentPin = new LibDrawPin();
+    CurrentPin = new LibDrawPin( CurrentLibEntry );
 
     CurrentDrawItem = CurrentPin;
     if( CurrentPin == NULL )
Index: kicad/eeschema/eeschema.cpp
===================================================================
--- kicad.orig/eeschema/eeschema.cpp	2009-06-25 05:36:52.000000000 -0300
+++ kicad/eeschema/eeschema.cpp	2009-06-25 05:37:28.000000000 -0300
@@ -68,8 +68,8 @@
 
 int       g_PlotFormat;                  /* flag = TYPE_HPGL, TYPE_PS... */
 int       g_PlotMargin;                  /* Marge pour traces du cartouche */
-float     g_PlotScaleX;
-float     g_PlotScaleY; /* coeff d'echelle de trace en unites table tracante */
+double    g_PlotScaleX;
+double    g_PlotScaleY; /* coeff d'echelle de trace en unites table tracante */
 
 HPGL_Pen_Descr_Struct g_HPGL_Pen_Descr;
 
@@ -93,6 +93,7 @@
 bool g_EditPinByPinIsOn = FALSE;
 
 int g_LibSymbolDefaultLineWidth; /* default line width  (in EESCHEMA units) used when creating a new graphic item in libedit : 0 = default */
+int g_DrawDefaultLineThickness; /* Default line (in EESCHEMA units) thickness */
 int g_DrawMinimunLineWidth;      /* Minimum line (in EESCHEMA units) thickness used to draw items on screen; 0 = single pixel line width */
 int g_PlotLine_Width;            /* Minimum line (in EESCHEMA units) thickness used to Plot/Print items */
 // Color to draw selected items
